core/tests/test_utils.py

```diff
@@ -483,12 +483,32 @@ def get_expected_logout_url(self, slug):
         """Returns the expected logout URL."""
         return current_user_services.create_logout_url(slug)
 
-    def get_response(self, url, params=None, expect_errors=False,
-                     expected_status_int=200,
-                     expected_content_type='text/html'):
-        """Get a response, transformed to a Python object."""
+    def get_response(
+            self, url, expected_content_type, params=None,
+            expected_status_int=200):
+        """Get a response, transformed to a Python object.
+
+        Args:
+            url: str. The URL to fetch the response.
+
+            expected_content_type: str. The content type to expect.
+
+            params: str or dict. A query string, or a dictionary that will be
```

Comment:
Looks like it could also be None too, right?

Can we also make this either str or dict (not both) depending on how _we_ use it? The webtest API may be more general but I think it's better for our codebase to be stricter and have a specific way of doing it.

Reply:
Thanks for checking! Looking at the link you sent, something seems wrong -- generally when csrf tokens are passed into get_json() it shouldn't happen like that, I think.

I would suggest looking at how other CSRF tokens are passed into get_json in other tests, and following a similar procedure here. Then, ensure that the docstring here only accepts a dict. It might be worth converting the query params that are in the URL of the test you linked into an actual params dict, too, for consistency, but that's optional.

Reply:
I can't understand why we need CSRF tokens in get requests. No other get_json function uses CSRF tokens in the current codebase so I don't know why its used here. Generally its used in put or post requests and I think its not required in the URL that I shared. What's your opinion on this?

Reply:
It's probably not needed in this case, then.